File: HorseRacing.Web/src/interfaces/api/contracts/model/game/responses/join-game-with-bet/join-game-with-bet-response.ts
import type { BaseResponse } from '../../../../base/responses/base-response';
import type { JoinGameWithBetResponseDto } from './join-game-with-bet-response-dto';

export class JoinGameWithBetResponse implements BaseResponse<JoinGameWithBetResponseDto> {
    Data: JoinGameWithBetResponseDto;

    constructor(data?: JoinGameWithBetResponseDto) {
        this.Data = data || { IsLastPlayer: false };
    }
}

File: HorseRacing.Contracts/Models/Game/Responses/JoinGameWithBet/JoinGameWithBetResponse.cs
using HorseRacing.Contracts.Base.Responses;

namespace HorseRacing.Contracts.Models.Game.Responses.JoinGameWithBet
{
    public class JoinGameWithBetResponse : BaseResponse<JoinGameWithBetResponseDto>
    {
        public JoinGameWithBetResponse() : base() { }
        public JoinGameWithBetResponse(JoinGameWithBetResponseDto data) : base(data) { }
    }
}
